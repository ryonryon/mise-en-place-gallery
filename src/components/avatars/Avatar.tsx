import styled from "@emotion/styled";
import * as React from "react";

interface Props extends React.ImgHTMLAttributes<HTMLImageElement> {
  src: string;
  height: string;
  width: string;
}
const Avatar = ({ src, height, width, ...rest }: Props) => {
  const StyledImg = styled.img`
    vertical=align: middle;
    height: ${height};
    width: ${width};
    border-radius: 50%;
  `;
  return <StyledImg src={src} {...rest} alt="avatar" />;
};

Avatar.defaultProps = {
  height: "50px",
  width: "50px"
};

export default Avatar;
