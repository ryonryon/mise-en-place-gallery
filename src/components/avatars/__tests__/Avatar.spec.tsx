import * as React from "react";
import * as renderer from "react-test-renderer";

import Avatar from "../Avatar";

describe("Avatar test", () => {
  it("tests Avatar renders correctly ", () => {
    const avatar = renderer.create(<Avatar src={"img"} />).toJSON();

    expect(avatar).toMatchSnapshot();
  });
});
