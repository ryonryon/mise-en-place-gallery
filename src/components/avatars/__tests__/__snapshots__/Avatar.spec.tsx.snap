// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar test tests Avatar renders correctly  1`] = `
<img
  alt="avatar"
  className="css-u2jqka"
  src="img"
/>
`;
